public/app/features/dashboard/components/ShareModal/SharePublicDashboard.tsx

```diff
@@ -50,7 +51,14 @@ export const SharePublicDashboard = (props: Props) => {
 
   return (
     <>
+      {!dashboardCanBePublic(props.dashboard) && (
```

Comment:
If the dashboard has a template variable added and the page is not reloaded, props.dashboard is not updated and this alert will not show up.

This should check against retrieved dashboard state instead of what is passed in via props.Dashboard.

Reply:
Completed as part of refactor.